epidose/device/wps_scanner_d.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/sh
+#
+# Check for available networks and connect to the closest one using the WPS PBS
+#
+# Copyright 2020 Diomidis Spinellis
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+set -e
+
+export APP_NAME=wps_scanner_d
+
+# Pick up utility functions relative to the script's source code
+UTIL="$(dirname "$0")/util.sh"
+
+# Source common functionality (logging, WiFi)
+# shellcheck source=epidose/device/util.sh
+SERVER_URL=none . "$UTIL"
+
+wps_connect()
+{
+  # Turn on the orange blinking of the WPS LED
+  #run_python device_io --orange-on
```

Comment:
You also need to 
* Wait for the button to be pressed 
* Indicate whether the connection succeeded or not (e.g. turn on Green or Red)
* Turn off the LED when done after a small pause.
* Keep in mind that the *watchdog* will also be flashing the LED.  Consequently, you need to inform the watchdog to pause its LED activities, perhaps by creating (and then removing) a file in `/var/lock/`.

Reply:
Regarding the first bullet point, I guess using the python_run device_io --wifi-wait will do the job, right?
About the last one, I guess you mean by using the `touch /var/lock/watchdog_d` will pause the execution of the corresponding daemon, right?
But, I do not see the watchdog daemon inside the setup.py or the Makefile.
Also, I cannot find it among my processes `$ ps -aux | grep epidose`.

Reply:
1. Yes
2. Yes, something that watchdog can monitor
3. Watchdog is running through *supervisor* and `setup.py`
```
$ sudo supervisorctl status
epidose:beacon_rx                RUNNING   pid 446, uptime 5 days, 5:58:33
epidose:beacon_tx                RUNNING   pid 19546, uptime 0:09:35
epidose:update_filter            RUNNING   pid 447, uptime 5 days, 5:58:33
epidose:upload_seeds             RUNNING   pid 448, uptime 5 days, 5:58:33
epidose:watchdog                 RUNNING   pid 450, uptime 5 days, 5:58:32
```

Reply:
This is what I get from watchdog's log:

`Traceback (most recent call last):
  File "/opt/venvs/epidose/lib/python3.7/site-packages/epidose/device/watchdog_d.py", line 154, in <module>
    main()
  File "/opt/venvs/epidose/lib/python3.7/site-packages/epidose/device/watchdog_d.py", line 142, in main
    v = get_battery_voltage()
  File "/opt/venvs/epidose/lib/python3.7/site-packages/epidose/device/device_io.py", line 156, in get_battery_voltage
    spi = spi_setup()
  File "/opt/venvs/epidose/lib/python3.7/site-packages/epidose/device/device_io.py", line 137, in spi_setup
    spi.open(0, 0)
FileNotFoundError: [Errno 2] No such file or directory`

(venv) pi@raspberrypi:~/epidose $ sudo supervisorctl status
epidose:beacon_rx                RUNNING   pid 22995, uptime 3:19:21
epidose:beacon_tx                RUNNING   pid 24070, uptime 0:02:15
epidose:update_filter            RUNNING   pid 22996, uptime 3:19:21
epidose:upload_seeds             RUNNING   pid 22997, uptime 3:19:21
epidose:watchdog                 FATAL     Exited too quickly (process log may have details)

I guess when I will have the actual device this will be solved.

Reply:
You're not running it on a real device, right?  Where is your Epidose package? Can you please coordinate to track it?

Reply:
I am tracking it every day. Its latest location it is the Athens Airmail Center.
I can see something similar in the Hungarian's postal tracking service.

Reply:
I run a couple of tests in order to see if the LED lights are flashing as expected.
Seems the `touch /var/lock/suspend-watchdog-led` does not stop the blinking status light.
I have also tried using the `supervisorctl stop epidose:watchdog` but the status light is still blinking.

Reply:
You can enable/add debugging to the watchdog in order to see what is going on.